apps/meteor/app/livechat/server/api/lib/users.ts

```diff
@@ -10,20 +11,62 @@ import { escapeRegExp } from '@rocket.chat/string-helpers';
 async function findUsers({
 	role,
 	text,
+	onlyAvailable = false,
+	excludeId,
+	showIdleAgents = true,
 	pagination: { offset, count, sort },
 }: {
 	role: IRole['_id'];
 	text?: string;
+	onlyAvailable?: boolean;
+	excludeId?: string;
+	showIdleAgents?: boolean;
 	pagination: { offset: number; count: number; sort: any };
 }): Promise<{ users: ILivechatAgent[]; count: number; offset: number; total: number }> {
-	const query = {};
+	const query: FilterOperators<ILivechatAgent> = {};
 	if (text) {
 		const filterReg = new RegExp(escapeRegExp(text), 'i');
 		Object.assign(query, {
 			$or: [{ username: filterReg }, { name: filterReg }, { 'emails.address': filterReg }],
 		});
 	}
 
+	if (onlyAvailable) {
+		Object.assign(query, {
```

Comment:
is `Object.assign` being used just to not mess up with types? 😬

Reply:
In a part, yeah 😬 
The other part is to keep consistent with the file doing these assigns instead of spread, since the whole thing is for a patch.

Don't mind changing tho